packages/components/src/autocomplete/list.js

```diff
@@ -0,0 +1,211 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import classnames from 'classnames';
+import { Component, createRef } from '@wordpress/element';
+import { isEqual } from 'lodash';
+import { ENTER, ESCAPE, UP, DOWN, LEFT, TAB, RIGHT } from '@wordpress/keycodes';
+import PropTypes from 'prop-types';
+
+/**
+ * A list box that displays filtered options after search.
+ */
+class List extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.handleKeyDown = this.handleKeyDown.bind( this );
+		this.select = this.select.bind( this );
+		this.optionRefs = {};
+		this.listbox = createRef();
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { filteredOptions } = this.props;
+
+		// Remove old option refs to avoid memory leaks.
+		if ( ! isEqual( filteredOptions, prevProps.filteredOptions ) ) {
+			this.optionRefs = {};
+		}
+	}
+
+	getOptionRef( index ) {
+		if ( ! this.optionRefs.hasOwnProperty( index ) ) {
+			this.optionRefs[ index ] = createRef();
+		}
+
+		return this.optionRefs[ index ];
+	}
+
+	select( option ) {
+		const { onSelect } = this.props;
+
+		if ( option.isDisabled ) {
+			return;
+		}
+
+		onSelect( option );
+	}
+
+	scrollToOption( index ) {
+		const listbox = this.listbox.current;
+
+		if ( listbox.scrollHeight > listbox.clientHeight ) {
+			const option = this.optionRefs[ index ].current;
+			const scrollBottom = listbox.clientHeight + listbox.scrollTop;
+			const elementBottom = option.offsetTop + option.offsetHeight;
+			if ( elementBottom > scrollBottom ) {
+				listbox.scrollTop = elementBottom - listbox.clientHeight;
+			} else if ( option.offsetTop < listbox.scrollTop ) {
+				listbox.scrollTop = option.offsetTop;
+			}
+		}
+	}
+
+	handleKeyDown( event ) {
+		const { filteredOptions, onChange, onSearch, selectedIndex } = this.props;
+		if ( filteredOptions.length === 0 ) {
+			return;
+		}
+
+		let nextSelectedIndex;
+		switch ( event.keyCode ) {
+			case UP:
+				nextSelectedIndex = null !== selectedIndex
+					? ( selectedIndex === 0 ? filteredOptions.length : selectedIndex ) - 1
+					: filteredOptions.length - 1;
+				onChange( nextSelectedIndex );
+				this.scrollToOption( nextSelectedIndex );
+				event.preventDefault();
+				event.stopPropagation();
+				break;
+
+			case TAB:
```

Comment:
The tab functionality in https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html brings focus to the next focusable element even when the options are visible.

Here, tab brings me to the next focusable element only when the options are hidden. In the case of the first example in dev docs, if you navigate to it using tab, you can't tab out of it unless something is selected.